src/libifconfig.c

```diff
@@ -401,6 +401,36 @@ ifconfig_get_capability(ifconfig_handle_t *h, const char *name,
 	return (0);
 }
 
+int ifconfig_get_groups(ifconfig_handle_t *h, const char *name,
+    struct ifgroupreq *ifgr)
+{
+	int s, len;
+
+	if (ifconfig_socket(h, AF_LOCAL, &s) != 0) {
+		return (-1);
+	}
+
+	memset(ifgr, 0, sizeof(*ifgr));
+	strlcpy(ifgr->ifgr_name, name, IFNAMSIZ);
+
+	if (ioctl(s, SIOCGIFGROUP, (caddr_t)ifgr) == -1) {
+		if (errno == EINVAL || errno == ENOTTY)
```

Comment:
It was impossible to use `ifconfig_ioctlwrap` here before 521f63336b0d9f27fd80f83a26240e15387fdf29, when I changed the prototype for that function.  The current version does use `ifconfig_ioctlwrap`.

It looks like the checks for EINVAL and ENOTTY were present when mlaier originally imported this code from OpenBSD in 2006.  It's still present in OpenBSD today.  But I can't guess why it's present there.  OpenBSD's `if_getgroup` looks like ours.  However, if we remove that line, we should remove it from ifconfig first, not libifconfig.